api/routes/user_routes.go

```diff
@@ -39,7 +42,10 @@ func NewUserRoutes(
 func (s *UserRoutes) Setup() {
 	s.logger.Info("Setting up routes")
 
-	api := s.handler.Group("/api")
+	api := s.handler.Group("/api").Use(func(ctx *gin.Context) {
+		s.authMiddleware.HandleRoles(ctx, constants.RoleIsAdmin)
```

Comment:
@Denes-cilwal `HandleRoles` does not look a good name. Let's change it to `HandleAuthWithRole`